src/Collections-Sequenceable/OrderedCollection.class.st

```diff
@@ -685,6 +685,23 @@ OrderedCollection >> removeAt: index [
 	^removed
 ]
 
+{ #category : #removing }
+OrderedCollection >> removeDuplicates [
+	| iterator |
+	"Remove the copies of elements, but keep the same order"
+	
+	self ifEmpty: [ ^ self ].
+	iterator := 1.
+	[ iterator <= self size ]
+		whileTrue: [ | each newIndex |
+			each := self at: iterator.			
```

Comment:
Here there is a possible optimisation (together with the code suggestion under) that allow to get better performance according to a benchmark I will attach in my main comment.
```suggestion
			each := self at: iterator.
			newIndex := iterator + 1.
```